contracts/Verifier.sol

```diff
@@ -338,25 +337,40 @@ contract Verifier is Ownable, HydratedRuntime {
             return;
         }
 
+        // left can not be zero
+        if (dispute.solver.left == bytes32(0)) {
+            enforcer.result(dispute.executionId, false, dispute.challengerAddr);
+            dispute.state |= CHALLENGER_VERIFIED;
+            return;
+        }
+        if (dispute.challenger.left == bytes32(0)) {
+            enforcer.result(dispute.executionId, true, dispute.challengerAddr);
+            dispute.state |= SOLVER_VERIFIED;
+            return;
+        }
+
         if (dispute.treeDepth == 1 && dispute.challenger.left != dispute.solver.left) {
             require(dispute.witness != bytes32(0));
-            // we may only require 6 or 2 witnesses
-            // this could be improved
-            require(witnesses.length == 6 || witnesses.length == 2);
+            // TODO: we could keep track of the last depth taken right,
```

Comment:
I suppose we could do this by:
- when go right, witness = left
- when go left, witness = witness.right, need to respond with (witness.left, witness.right) to verify.

Reply:
@peara True 😺. Updated 😏